source/appModules/soffice.py

```diff
@@ -350,6 +352,40 @@ def event_selectionWithIn(self):
 			curFocus.announceSelectionChange()
 
 
+class SymphonyButton(IAccessible):
+
+    def event_stateChange(self):
+        # announce new state of toggled toolbar button to indicate formatting change
+        # if registered gesture resulted in button state change
+        if (
+            SymphonyDocument.announce_toolbar_button_toggle
+            and self.parent
+            and self.parent.role == controlTypes.Role.TOOLBAR
+            and time.time()
+            < (
+                SymphonyDocument.last_formatting_gesture_event_time
+                + SymphonyDocument.GESTURE_ANNOUNCEMENT_TIMEOUT
+            )
+        ):
+            states = self.states
+            enabled = (
+                controlTypes.State.PRESSED in states
+                or controlTypes.State.CHECKED in states
+            )
+            # button's accessible name is the font attribute, e.g. "Bold", "Italic"
+            if enabled:
+                # Translators: a message when toggling formatting (e.g. bold, italic) in LibreOffice
+                message = _("{textAttribute} on").format(textAttribute=self.name)

```

Comment:
Is `self.name` translated in non-English UI of LO?

Reply:
Yes. I just tested with German UI, and pressing Ctrl+B causes the announcement of "Blocksatz on".  This is not the equivalent of "Bold", but "Justified", as the German keyboard shortcuts are different.
From what I could see, taking care that the right keyboard shortcuts/gestures are handled is generally a responsibility of the corresponding translation (i.e. language communities). Is that correct?

Reply:
Yes until now, localization of shortcuts are handled by translators.

This may change or not change in a near future since the translation process is currently being modified.

But until NV Access gives further instructions, let's stick to the old method.